src/Common/BlechString.fs

```diff
@@ -0,0 +1,490 @@
+    //                        tabIndent <- Some <| tabs line
+    //                    elif not (tabs line = Option.get tabIndent) then
+    //                        unbalancedTabIndent <- true
+                        
+    //                    // determine common identation
+    //                    if Option.isNone commonIndent then // init commonIdent
+    //                        printfn "init ident:>>%s<<" indent
+    //                        commonIndent <- Some indent
+    //                    else
+    //                        let ci = Option.get commonIndent
+    //                        if ci.StartsWith indent then
+    //                            commonIndent <- Some indent
+    //                        elif line.StartsWith ci then
+    //                            ()
+    //                        else
+    //                            commonIndent <- Some ""
+            
+    //        (unbalancedTabIndent, Option.defaultValue NoIndent commonIndent)
+
+    let isWhitespaceLine line = 
+        Regex.IsMatch(line, WhitespaceOnly)
+ 
+    let isEmptyLine line = 
+        String.Empty = line
+
+    //let getExtraWhitespaceLength line =
+    //    Regex.Match(line, WhitespaceOnly).Length
+
+    let getTabIndentation line = 
+        Regex.Match(line, TabIndentation).Length
+
+    let getIndentation line = 
+        Regex.Match(line, Indentation).Length
+
+    let findUnbalancedTabIndentations (lines: string seq) =
+        let mutable tabIndent = None
+        let unbalancedTabIndents = 
+            seq { for i, line in Seq.indexed <| Seq.tail lines do  // The first line cannot contain tabs due to the lexer
+                    if not (isEmptyLine line) then // a empty line is always balanced
+                        let lineTabIndent = getTabIndentation line
+                        if Option.isNone <| tabIndent then
+                            tabIndent <- Some (i, lineTabIndent) // the first tab indentation defines the standard 
+                            // yield i, None
+                        elif lineTabIndent <> snd (Option.get tabIndent) then
+                            // yield i, None
+                        //else
+                            yield i, lineTabIndent }
+        (tabIndent, unbalancedTabIndents)
+
+    //let findExtraWhitespace (lines : string seq) =
+    //    seq { for i, line in Seq.indexed lines do
+    //            if i > 1 && i = Seq.length lines - 1 then 
+    //                // the last of 2 or more lines may contain extra whitespace
+    //                yield None
+    //            else
+    //                let wsLen = getExtraWhitespaceLength line 
+    //                if wsLen = 0 then
+    //                    yield None
+    //                else
+    //                    yield Some wsLen }
+
+    // assumes balanced tab indentation and no extra whitespace in empty lines
+    let getCommonIndentation (lines: string seq) = 
+        let mutable commonIndent = None
```

Comment:
why not work with `mutable commonIdent = 0` ?
The logic of this function does not differentiate between `Some 0` and `None`